File: Data Structures & Algorithms/jump-game/submission-0.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        
        i = 0
        j = i + nums[i]
        while i < len(nums) and j < len(nums) - 1:
            i += 1
            if i > j:
                return False 
            j = max(i+nums[i], j)
        
        
        return True
            
        

File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& nums) {
        int max_area = 0;
        int i = 0;
        int j = nums.size()-1;

        while(i<j){
            int width = j - i;
            int height = min(nums[i], nums[j]);
            int curr_area = height*width; 
            cout << " width: " << width << ", height: " << height << ", area: " << curr_area << endl;
            max_area = max(curr_area, max_area);

            if (nums[i] < nums[j]){
                i++;
            }
            else if(nums[i] > nums[j]){
                j--;
            }
            else{
                i++; j--;
            }
        }

        return max_area;
        
    }
};
